File: trunk/random-coverage/sim-single.pl
#!/usr/bin/env perl

# Copyright (C) 2011 Tianyang Li
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License

# simluate uniform random covering of a
# transcript by reads of equal length
#
# this is a model for Illumina single end reads

use strict;
use warnings;

use Getopt::Std;

# -L	length of transcript
# -N	number of reads
# -k 	length of read
# -t	number of threads
my %params = ( t => "1" );
getopts( 'L:N:k:t:', \%params );

# calculate probability using my formula
{
	use bignum;

	sub calc_prob {
		my $params = shift;
		my $k      = $params->{k};
		my $N      = $params->{N};
		my $L      = $params->{L} - $k + 1;    # effective length
		my $prob = 0;
		$prob;
	}

	printf "%g\n", &calc_prob( \%params );
}

File: trunk/random-coverage/sim_single.py
#!/usr/bin/env python

# Copyright (C) 2011 Tianyang Li
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License

import sys
import scipy
import multiprocessing
import getopt

def main(sysargs):
    try:
        opts, args = getopt.getopt(sysargs, 'L:N:k:r:p:')
    except getopt.GetoptError as err:
        print >> sys.stderr, str(err)
        sys.exit(2)
    L = None # effective length
    k = None
    N = None
    runs = None
    p = 1
    for o, a in opts:
        if o == '-L':
            eL = int(a)
        elif o == '-k':
            k = int(a)
        elif o == '-r':
            runs = int(a)
        elif o == '-N':
            N = int(a)
        elif o == '-p':
            p = int(a)
    if (eL == None) or (k == None) or (N == None) or (runs == None):
        print >> sys.stderr, "Missing options!"
        sys.exit(2)
    L = L - k + 1 
    
if __name__ == '__main__':
    main(sys.argv[1:])
